tests/dummy/app/serializers/component.js

```diff
@@ -1,3 +1,21 @@
 import JSONAPISerializer from '@ember-data/serializer/json-api';
 
-export default class extends JSONAPISerializer {}
+export default class extends JSONAPISerializer {
+  serialize(snapshot, ...args) {
+    const data = super.serialize(snapshot, ...args);
+    const { record } = snapshot;
+
+    if (data.data?.attributes) {
+        // NOTICE: Remove all the unchanged attributes in the payload.
+        const changedAttributes = Object.keys(record.changedAttributes());
+
+        Object.entries(data.data.attributes).forEach(([attributeName, value]) => {
+          if (!changedAttributes.includes(attributeName)) {
+            delete data.data.attributes[attributeName];
+          }
+        });
+    }
```

Comment:
What is this for?

Reply:
It's written in the comment above `// NOTICE: Remove all the unchanged attributes in the payload.`

Reply:
Why is this required for the test?

Reply:
It's what the test is about, checking that the changedAttributes is correct between the `save` and the return of the `save` (So between `willCommit` and `didCommit`)